api/models/Post.js

```diff
@@ -150,11 +156,32 @@ module.exports = bookshelf.Model.extend({
   },
 
   pushCommentToSockets: function (comment) {
-    this.sendToSubscribedSockets('commentAdded', comment)
+    var postId = this.id
+    this.sendToSubscribedSockets(`posts/${postId}`, 'commentAdded', comment)
+  },
+
+  pushMessageToSockets: function (message, userIds) {
+    var postId = this.id
+    const excludingSender = userIds.filter(id => id !== message.user_id.toString())
+    if (this.get('num_comments') === 1) {
+      excludingSender.forEach(id => this.pushSelfToSocket(id)) 
+    } else {
+      excludingSender.forEach(id => this.sendToSubscribedSockets(`users/${id}`, 'messageAdded', { postId, message })) 
+    }
   },
 
   pushTypingToSockets: function (userId, userName, isTyping, socketToExclude) {
-    this.sendToSubscribedSockets('userTyping', { userId, userName, isTyping }, socketToExclude)
+    var postId = this.id
+    this.sendToSubscribedSockets(`posts/${postId}`, 'userTyping', { userId, userName, isTyping }, socketToExclude)
+  },
+
+  pushSelfToSocket: function (userId) {
+    var that = this
+    const opts = { withComments: 'all' }
```

Comment:
style note: i've been putting spaces inside curly brackets only when they're being used for destructuring.